pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -251,6 +251,10 @@ func runArgsForNode(node *config.Node, clusterIPFamily config.ClusterIPFamily, n
 	}
 	args = append(args, mappingArgs...)
 
+	if node.Gpus != "" {
+		args = append(args, "--gpus", node.Gpus)
```

Comment:
what does this flag do vs passing in the device as an extraMount path, given the node already has full capabilities?

Reply:
This docker flag seems to be fairly poorly documented since its inception many months ago, so believe me I wish I had a better answer for you.

It appears that as of recent docker versions, forcing the use of nvidia's `nvidia-container-runtime` runc wrapper is no longer sufficient in itself to expose identified GPU devices to containers under that runtime. This flag appears to be recently required (as of 19.03.13?) in addition the nv runtime configuration in order to present the identified devices to containers under docker. As best as I can tell, exposing devices via mount alone does not appear to be a supported path to obtain the necessary NVML libs and utils.